docs/quickstart/clustering.md

```diff
@@ -0,0 +1,129 @@
+from sklearn import metrics
+
+def score(X, n_clusters):
+    model = KMeans(n_init="auto", n_clusters=n_clusters, random_state=1)
+    model.fit(X)
+    predicted = model.predict(X)
+    return {"n_clusters": n_clusters,
+            "silhouette_score": metrics.silhouette_score(X, predicted),
+            "calinski_harabasz_score": metrics.calinski_harabasz_score(X, predicted),
+            "davies_bouldin_score": metrics.davies_bouldin_score(X, predicted)}
+
+
+df_metrics = pd.DataFrame(score(X, n_clusters) for n_clusters in (2, 3, 4, 5, 6, 7, 8, 9, 10))
+df_metrics.set_index("n_clusters", inplace=True)
+
+(df_metrics.style
+ .highlight_max(subset=["silhouette_score", "calinski_harabasz_score"], color="lightgreen")
+ .highlight_min(subset=["davies_bouldin_score"], color="lightgreen"))
+```
+
+All three metrics have their *best* value when `n_clusters=5`. We know this is the best value since our data has 5 clusters; however, when using real datasets, you might find that these metrics might not agree, so it's advisable to understand how each metric is computed and choose the best one for your project.
+
+You can also find the best number of clusters visually. Let's see how to do it using an elbow curve.
+
++++
+
+## Optimal number of clusters
+
+### `plot.elbow_curve`
+
+```{important}
+Currently, `plot.elbow_curve` only works with the following sklearn models: `KMeans`, `BisectingKMeans`, and `MiniBatchKMeans`
+```
+
+An elbow curve evaluates the sum of squared errors (i.e., how far each point is from its assigned cluster center). So, naturally, the more centers you have, the lower this metric will be. However, a good clustering model does not necessarily minimize this metric but balances the trade-off between lowering the sum of squared errors and having a small number of clusters (since those two are competing objectives).
+
+The elbow curve will plot the sum of squared errors for a different number of clusters; then, you can visually evaluate when increasing the number of clusters is not worth it, as it'll not yield a significant decrease in the sum of squared errors.
+
+```{code-cell} ipython3
+from sklearn_evaluation import plot
+from sklearn import cluster
+
+model = KMeans(n_init="auto", random_state=1)
+```
+
+```{code-cell} ipython3
+_ = plot.elbow_curve(X, model, n_clusters=(2, 3, 4, 5, 6, 7, 8))
+```
+
+In our curve, we see significant improvements when moving from 2 to 5 clusters; but increasing to 6 or larger does not yield substantial improvements; hence, we can conclude that 5 is the optimal number of clusters.
+
++++
+
+### `plot.silhouette_analysis`
+
+We can visually represent the `silhouette_score` to assess the number of clusters. Remember that values close to +1 indicate that the clusters are well-separated. Another characteristic to consider is the size of each silhouette plot. If they're too different, it means some clusters are tiny while others are too large (see, for example, `n_clusters=6,7,8`).
```

Comment:
Maybe worth reversing the order so the tiny clusters pop out immediately.

Reply:
reversing the order where? in the description? 

cause if you mean the plot, that depends on the model output

Reply:
maybe the problem is that my explanation was not clear enough. I updated it, let me know what you think

Reply:
let me check.

Reply:
Yeah I think it's better now, my original point was show the 8-6 first so it's super clear which ones are faulty, but I think stating 6-8 is good enough.